From d8f58f77e7a595db4a4dabf1a1ec0cf4e5145167 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenko@redhat.com>
Date: Sun, 21 May 2017 19:05:34 +0200
Subject: [PATCH] bindings: do not leak Repo in __next__()

We don't create new object in __getitem__, why should we do it in __next__?

Closes: https://github.com/openSUSE/libsolv/issues/207
Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>
---
 bindings/solv.i | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bindings/solv.i b/bindings/solv.i
index 96e5d23..5736f20 100644
--- a/bindings/solv.i
+++ b/bindings/solv.i
@@ -2609,7 +2609,6 @@ rb_eval_string(
 #ifdef SWIGPERL
   perliter(solv::Pool_repo_iterator)
 #endif
-  %newobject __next__;
   Repo *__next__() {
     Pool *pool = $self->pool;
     if ($self->id >= pool->nrepos)
-- 
2.13.0

